modules/detect_target/detect_target.py

```diff
@@ -15,23 +16,35 @@ class DetectTarget:
     Contains the YOLOv8 model for prediction
     """
 
-    def __init__(self, model_path: str):
-        self.model = ultralytics.YOLO(model_path)
+    def __init__(self, model_path: str, save_name: str=""):
+        self.__model = ultralytics.YOLO(model_path)
+        self.__counter = 0
+        self.__filename_prefix = save_name
 
     def run(self, data: frame_and_time.FrameAndTime) -> "tuple[bool, np.ndarray | None]":
         """
         Returns annotated image
         TODO: Change to PointsAndTime
         """
         image = data.frame
-        predictions = self.model.predict(image, stream=False)
+        predictions = self.__model.predict(
+            source=image,
+            half=True,
+            device=0,
+            stream=False
+        )
 
         if len(predictions) == 0:
             return False, None
 
         # TODO: Change this to image points for image and telemetry merge for 2024
         # (bounding box conversion code required)
-        image_annotated = predictions[0].plot(show_conf=True)
+        image_annotated = predictions[0].plot(conf=True)
+
+        # Logging
+        if self.__filename_prefix != "":
```

Comment:
Can we also log the bounding box to a txt file with the same name as the corresponding image. Reason being @YMRTZ mentioned that he needed a data for the VMGM.

Reply:
Done.